### PR TITLE
[tools] Validate and format/sort strings files before generation of localizations

### PR DESCRIPTION
**Pull request by pastk**
If a file is not valid then the process is terminated with an error message and no localization files are generated.

Needed-by: #1703

### CODE REVIEW

File: tools/unix/generate_localizations.sh [hunk 1]
```diff
@@ -42,6 +42,12 @@ if [ ! -f "$TWINE_PATH/$TWINE_GEM" ] || ! gem list -i twine; then
   )
 fi
 
+# Validate and format/sort strings files.
+STRINGS_UTILS="$OMIM_PATH/tools/python/strings_utils.py"
+$STRINGS_UTILS --validate --output
```
Comment (biodranik) [inline tools/unix/generate_localizations.sh:47]: Лучше в кавычки взять.

**Review by biodranik — APPROVED** [review]

**Comment by pastk** [comment]
We can also run `clean_strings_txt.py -s` just before generation of localizations to remove all translation keys which are no longer used in the code.
If automatic deletion sounds too scary we can add "unused translations" validation check, e.g. terminate if some translations are no longer used, so that a human will double check and run `clean_strings_txt.py -s` manually.
What do you think?

(and later we can improve `clean_strings_txt.py` to add new translation keys from the codebase which are not present in strings.txt)

**Comment by biodranik** [comment]
What if this script fails and says "please clear unused strings"?

File: tools/unix/generate_localizations.sh (hunk 1 above)
Comment (pastk) [reply tools/unix/generate_localizations.sh:47]: Я попробовал все виды кавычек, выдаёт
> tools/unix/generate_localizations.sh: line 47: tools/unix/../../tools/python/strings_utils.py --validate --output: No such file or directory

?

**Comment by pastk** [comment]
> What if this script fails and says "please clear unused strings"?

Yeap, exactly

File: tools/unix/generate_localizations.sh (hunk 1 above)
Comment (biodranik) [reply tools/unix/generate_localizations.sh:47]: Это точно на бранче где переименование было?

Comment (pastk) [reply tools/unix/generate_localizations.sh:47]: точно-точно
убираю кавычки - работает
добавляю - ошибка
try `"echo $HOME"` in bash

Comment (biodranik) [reply tools/unix/generate_localizations.sh:47]: `"$STRINGS_UTILS" --validate --output` не работает? Не надо всю строку в кавычки, конечно же. Параметры тогда не будут параметрами.

Comment (pastk) [reply tools/unix/generate_localizations.sh]: всё заработало, спасибо!

**Review by biodranik — APPROVED** [review]

**Pull request merged**